### PR TITLE
Added "null" and "undefined" expectations

### PR DESCRIPTION
**Pull request by pbihler**
To allow testing for the absence of properties

### CODE REVIEW

File: test/test.js [hunk 1]
```diff
@@ -133,4 +133,34 @@ describe('chai-shallow-deep-equal', function() {
           'Expected "42" to equal ' + '"2014-09-29T20:00:00.000Z" at path ".".'
       );
     });
+
+    it('success on missing properties', function() {
+        new chai.Assertion({a: 10}).to.be.shallowDeepEqual({a: 10, b: null});
```
Comment (michelsalib) [inline test/test.js]: `null` and `undefined` are different concept, can you elaborate a workflow where this kind of test would make sense?

**Comment by michelsalib** [comment]
Thanks for you PR @pbihler. Maybe `undefined` expectation makes more sense?

**Comment by pbihler** [comment]
Yes, you're right. I refactored the code, so "null" and "undefined" are both handled individually

File: chai-shallow-deep-equal.js [hunk 2]
```diff
@@ -23,6 +23,25 @@
...
+              throw 'Expected "' + actual +'" to equal "'+ expect +'" at path "'+ path +'".';
+            }
+
+            return true;
+        }
+
+        // undefined value
+        if (typeof expect == 'undefined') {
+            if (! (typeof actual == 'undefined')) {
+              throw 'Expected "' + actual +'" to equal "'+ expect +'" at path "'+ path +'".';
```
Comment (michelsalib) [inline chai-shallow-deep-equal.js]: Maybe replace with "to be undefined"

**Comment by michelsalib** [comment]
After this small label change we can merge.

**Comment by pbihler** [comment]
Sounds good. "to be null" as well...

**Comment by michelsalib** [comment]
Thanks @pbihler

**Comment by michelsalib** [comment]
Released in 1.1.0

**Pull request merged**